spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/registry/UriRegistryPopulator.java

```diff
@@ -73,6 +72,15 @@ public void setResourceLoader(ResourceLoader resourceLoader) {
 				for (String key : properties.stringPropertyNames()) {
 					try {
 						URI uri = new URI(properties.getProperty(key));
+						if (!overwrite) {
+							try  {
+								registry.find(key);
+							}
+							catch (IllegalArgumentException e) {
+								// this key already exists
```

Comment:
It looks like the `IllegalArgumentException` is thrown only when the key is not found. Hence, I think we need to register in that case instead of `continue` here and if the key is found then we can continue.